scripts/verify-mdlint.sh

```diff
@@ -0,0 +1,45 @@
+#! /bin/bash
+# vim: set ts=4 sw=4 et :
+
+# Usage: pre-commit.sh [--require-all]
+#   --require-all  Fail instead of warn if a checker is not found
+
+set -e
+
+# Run checks from root of the repo
+scriptdir="$(dirname "$(realpath "$0")")"
+cd "$scriptdir/.."
+
+# run_check <file_regex> <checker_exe> [optional args to checker...]
+# Pass empty regex when no regex is needed
+function run_check() {
+    regex="$1"
+    shift
+    exe="$1"
+    shift
+
+    if [ -x "$(command -v "$exe")" ]; then
+      if [ -z "$regex" ]; then
+          "$exe" "$@"
+      else
+          find . -path ./vendor -prune -o -regextype egrep -iregex "$regex" -print0 |
+              xargs -0rt -n1 "$exe" "$@"
+      fi
+    elif [ "$all_required" -eq 0 ]; then
+        echo "Warning: $exe not found... skipping some tests."
+    else
+        echo "FAILED: All checks required, but $exe not found!"
+        exit 1
+    fi
+}
+
+all_required=0
+if [ "x$1" == "x--require-all" ]; then
+    all_required=1
+fi
+
+# markdownlint: https://github.com/markdownlint/markdownlint
+# https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
+# Install via: gem install mdl
+run_check '.*\.md' mdl --style scripts/mdl-style.rb
```

Comment:
Is this script expected to be invoked while the "release-tools" directory is the current working directory? Then `find .` above will only check `*.md` files in that directory.

It probably will be necessary to locate the `scripts/mdl-style.rb` relative to the script itself, to cover both scenarios (testing in `csi-release-tools` and in some repo which has imported it).

Reply:
This can be enhanced to run  shell check  also ex https://github.com/ceph/ceph-csi/blob/cae8f529dc0d864c1677750320997e5553e45864/scripts/lint-extras.sh#L38-L49

Reply:
We already have support for shellcheck in csi-release-tools.

Reply:
@pohly couple of thoughts, right now, that invokes on the root directory and `style` is fetched from `scripts` dir which is one level down. If the intension is to invoke it on the directories where release tools are imported,  we could consider that as well. Is that what we want?

Reply:
The whole purpose of `csi-release-tools` is to provide shared supporting code for the other repos, so yes, we very much want to invoke Markdown checking in the other repos. It's much more important than checking the Markdown files in `csi-release-tools`.

It's usually easier to prototype future commits for `csi-release-tools` in some other repo first. Just modify `release-tools` there, commit, do a test PR. The PR will fail the "is release-tools dir modified" check, so it won't be merged accidentally, but a new check can be tested and proven to work in the environment were it will run later. Once that works, move the commit(s) to a PR for `csi-release-tools`. The README.md in this repo has instructions for that.